### PR TITLE
Solidity version bump to 0.8.8

### PR DESCRIPTION
**Pull request by kautukkundan**
**What does this pull request do? Explain your changes. (required)**

upgrades solidity version to 0.8.8 for all the protocol contracts and test/mock contracts. Also edited a few tests in case the something changed. For SafeMath, I simply included an upgraded version of the library instead of using the native solidity math features. I felt that this was out of the scope of this PR and if needed could be done separately.

I have tried to combine related contracts together in a single commit for ease in review process and I suggest going over commits individually. 

Most code changes are referenced from an earlier exercise at https://github.com/livepeer/protocol/pull/465

**Specific updates (required)**

- check commit history

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

related to https://github.com/code-423n4/2022-01-livepeer-findings/issues/10

**Checklist:**

- [x] README and other documentation updated
- [x] All tests using `yarn test` pass

### CODE REVIEW

File: contracts/bonding/BondingManager.sol
Comment (yondonfu) [inline contracts/bonding/BondingManager.sol]: Why does there need to be an `override` here? AFAIK as of solc 0.8.8 contracts that inherit from interfaces do not need to specify the `override` keyword when overriding functions defined in the interface.

File: contracts/IController.sol
Comment (yondonfu) [inline contracts/IController.sol]: Could the visibility of `getContract()` stay as `public` here since the implementation in `Controller` is public?

File: contracts/pm/TicketBroker.sol
Comment (yondonfu) [inline contracts/pm/TicketBroker.sol]: Does the ordering matter here?

File: contracts/pm/mixins/MixinContractRegistry.sol
Comment (yondonfu) [inline contracts/pm/mixins/MixinContractRegistry.sol]: If we're keeping around MContractRegistry anyway could we just inherit from that abstract contract which already imports the interfaces above which would reduce the diff here?

File: contracts/pm/mixins/MixinReserve.sol
Comment (yondonfu) [inline contracts/pm/mixins/MixinReserve.sol:8]: Thoughts on continuing to inherit from MContractRegistry instead so this abstract contract doesn't need to be aware of the impl of the functions in MixinContractRegistry and those implementations would just be mixed in via the MixinContractRegistry mixin in TicketBroker? Mainly interested in whether we can minimize the diff here if this change is not needed.

File: contracts/pm/mixins/MixinTicketBrokerCore.sol
Comment (yondonfu) [inline contracts/pm/mixins/MixinTicketBrokerCore.sol]: Thoughts on continuing to inherit from MTicketProcessor instead so this abstract contract doesn't need to be aware of the impl of the functions in MixinTicketProcessor and those implementations would just be mixed in via the MixinTicketProcessor mixin in TicketBroker? Mainly interested in whether we can minimize the diff here if this change is not needed.

File: contracts/pm/mixins/MixinTicketProcessor.sol
Comment (yondonfu) [inline contracts/pm/mixins/MixinTicketProcessor.sol]: Same comment here about inheriting from MContractRegistry & MTicketProcessor as the other comments about inheriting from the M prefixed contracts instead of the Mixin prefixed contracts with concrete implementations.

File: contracts/pm/mixins/MixinWrappers.sol
Comment (yondonfu) [inline contracts/pm/mixins/MixinWrappers.sol]: Same comment here about inheriting from MContractRegistry & MTicketBrokerCore as the other comments about inheriting from the M prefixed contracts instead of the Mixin prefixed contracts with concrete implementations.

**Review by yondonfu — CHANGES_REQUESTED** [review]
The commit structure was *very* helpful for review - thanks!

Overall looks solid and just left a few small comments.

A note about the current mixin contract architecture used for the TicketBroker since there are a few comments around the changes there. Originally the thought process behind the mixin contract architecture was that it would be helpful in breaking up the functionality of a large contract into smaller contracts. The architecture ended up with the some of the smaller mixin contracts calling functions defined in other mixin contracts so there was an attempt to decouple things by having an individual mixin contract only inherit an "interface mixin contract" (i.e. the M prefixed contracts) to avoid every mixin contract needing to be aware of the implementation of functions that are supposed to be defined in other mixin contracts. TBH, I'm not a huge fan of how this turned out at this point and I think the separation of logic could be cleaner. But, for now since we're not planning on making meaningful changes to the architecture, I'm in favor of minimizing the code diff here and sticking with that architecture if possible.

File: contracts/IController.sol
Comment (kautukkundan) [reply contracts/IController.sol]: my bad, I first tried converting the contract to an interface when I added the external keyword, then did not change it back to public when I made it abstract again. fixed in ebbb1ef

File: contracts/bonding/BondingManager.sol
Comment (kautukkundan) [reply contracts/bonding/BondingManager.sol]: my bad, did not change it after converting IBondingManager from abstract to interface. fixed in ca65491

**Comment by kautukkundan** [comment]
Will squash fixups before merge once 69ac6cf 51c2bff are good to go

**Review by yondonfu — APPROVED** [review]
LGTM after squashing the fixup commits

**Pull request merged**
